### PR TITLE
Minor fixes to the model

### PR DESCRIPTION
**Pull request by ariG23498**
Fixes include:
1. Initialization of `dropout` for each residual path
2. Initializing the `nn.ModuleList()` in the beginning and later adding the modules to it directly.

### CODE REVIEW

File: src/llm/wmt_model.py [hunk 1]
```diff
@@ -111,13 +114,11 @@ def __init__(
             f"model_dim was {model_dim} and num_heads was {num_heads}"
         )
         dim_k = dim_v = model_dim // num_heads
-        self.projection_heads = list()
+        self.projection_heads = nn.ModuleList()
```
Comment (ayulockin) [inline src/llm/wmt_model.py:117]: Thanks for opening this PR but a few of this is already addressed in #8. Can you take a look? I have not yet completed it but I can open it up for review. I am implementing the train logic there so we can tag team.

File: src/llm/wmt_model.py [hunk 2]
```diff
@@ -153,21 +154,23 @@ def __init__(
         )
         self.layer_norm1 = nn.LayerNorm(normalized_shape=model_dim)
         self.layer_norm2 = nn.LayerNorm(normalized_shape=model_dim)
-        self.residual_dropout = ResidualDropout(dropout_rate=dropout_rate)
+        self.residual_dropout_1 = ResidualDropout(dropout_rate=dropout_rate)
+        self.residual_dropout_2 = ResidualDropout(dropout_rate=dropout_rate)
```
Comment (ayulockin) [inline src/llm/wmt_model.py:158]: nice catch. we needed two separate dropout layers.

File: src/llm/wmt_model.py [hunk 3]
```diff
@@ -324,10 +328,11 @@ def __init__(self, model_dim: int, max_seq_len: int, dropout_rate: float):
         self.residual_dropout = ResidualDropout(dropout_rate=dropout_rate)
 
     def forward(
-        self, inputs: Float[Tensor, "batch seq_len model_dim"]  # type: ignore
+        self,
+        inputs: Float[Tensor, "batch seq_len model_dim"],  # type: ignore
     ) -> Float[Tensor, "batch seq_len model_dim"]:  # type: ignore
         x = inputs + self.position_encoding[: inputs.size(1)]
-        x = self.residual_dropout(x)
+        x = self.residual_dropout(x) # this is just dropout of x (no residuals)
```
Comment (ayulockin) [inline src/llm/wmt_model.py:335]: thanks for adding this comment.

File: src/llm/wmt_model.py (hunk 1 above)
Comment (ariG23498) [reply src/llm/wmt_model.py:117]: Maybe we can merge this and rebase #8 and open up for review?

Comment (ayulockin) [reply src/llm/wmt_model.py:117]: I have made a few changes in #8. I already added the minor fixes you added here into that PR. 🥲

Comment (ayulockin) [reply src/llm/wmt_model.py:117]: Sorry I should have informed you sooner.

Comment (ariG23498) [reply src/llm/wmt_model.py:117]: How would you like me to progress here? I am open to closing this PR in favour of #8

Comment (ayulockin) [reply src/llm/wmt_model.py:117]: If the changes in 8 matches with your PR, let's close this one.

Comment (ariG23498) [reply src/llm/wmt_model.py:117]: Cool, feel free to close this PR.
